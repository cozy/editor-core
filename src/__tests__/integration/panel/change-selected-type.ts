import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  editable,
  expectToMatchDocument,
  fullpage,
  quickInsert,
} from '../_helpers';
import {
  goToEditorTestingExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';
import { selectors } from './_utils';
import { PanelSharedCssClassName } from '@atlaskit/editor-common';

BrowserTestCase(
  'change-selected-type.ts: Select panel and then change type',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
      allowPanel: true,
    });

    await page.click(fullpage.placeholder);
    await quickInsert(page, 'Info panel');
    await page.waitForSelector(selectors.PANEL_EDITOR_CONTAINER);

    await page.type(editable, 'this text should be in the panel');

    // Select the left margin of the panel, selecting the node
    await page.click(`.${PanelSharedCssClassName.icon}`);

    // Change panel type to Error
    const selector = `[aria-label="Error"]`;
    await page.click(selector);

    await expectToMatchDocument(page, testName);
  },
);
