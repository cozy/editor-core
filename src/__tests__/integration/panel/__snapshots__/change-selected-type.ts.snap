// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change-selected-type.ts: Select panel and then change type 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "panelType": "error",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "text": "this text should be in the panel",
              "type": "text",
            },
          ],
          "type": "paragraph",
        },
      ],
      "type": "panel",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;
